src/actor/Verifier.ts

```diff
@@ -0,0 +1,148 @@
+    rawAttribute.push('claim.cTypeHash')
+    if (typeof legitimations !== 'undefined' && legitimations) {
+      rawAttribute.push('legitimation')
+    }
+    if (typeof delegationId !== 'undefined' && delegationId) {
+      rawAttribute.push('delegationId')
+    }
+    this.builder.requestPresentation({
+      requestedAttributes: rawAttribute,
+      reqUpdatedAfter,
+    })
+    this.ctypes.push(ctypeHash)
+    return this
+  }
+
+  /**
+   * [ASYNC] Concludes the presentation request.
+   *
+   * @param allowPE A boolean representing whether the verifier accepts a privacy enhanced presentation.
+   * @returns A session and a message object.
+   * The **session** object will be used in [[verifyPresentation]] and should be kept private by the verifier.
+   * The **message** object should be sent to the Claimer and used in [[createPresentation]].
+   */
+  public async finalize(
+    allowPE: boolean
+  ): Promise<[gabi.CombinedVerificationSession, IRequestClaimsForCTypes]> {
+    const { session, message } = await this.builder.finalise()
+    return [
+      session,
+      {
+        type: MessageBodyType.REQUEST_CLAIMS_FOR_CTYPES,
+        content: {
+          ctypes: this.ctypes,
+          peRequest: message,
+          allowPE,
+        },
+      },
+    ]
+  }
+}
+
+/**
+ * Initiates a verification by creating a request on the Verifier's side.
+ *
+ * @returns A [[PresentationRequestBuilder]] based on a [[CType]] and a list of required disclosed attributes of the [[Credential]]s.
+ */
+export function newRequest(): PresentationRequestBuilder {
+  return new PresentationRequestBuilder()
+}
+
+/**
+ * [ASYNC] Verifies the Claimer's presentation of [[Credential]]s.
+ *
+ * @param presentation The Claimer's presentation of the [[Credential]]s that should be verified, the result of [[createPresentation]].
+ * @param session The Verifier's private verification session created in [[finalize]].
+ * @param latestAccumulators The list of the latest accumulators for each [[Attester]] which signed a [[Credential]] of this presentation.
+ * @param attesterPubKeys The privacy enhanced public keys of all [[AttesterIdentity]]s which signed the [[Credential]]s.
+ * @returns An array representing
+ * **at index 0** whether the [[Credential]]s could be verified
+ * and **at index 1** an array of [[Claim]]s restricted on the disclosed attributes selected in [[requestPresentationForCtype]].
+ */
+export async function verifyPresentation(
+  presentation: ISubmitClaimsForCTypes,
+  session?: gabi.CombinedVerificationSession,
+  latestAccumulators?: gabi.Accumulator[],
+  attesterPubKeys?: PublicAttesterIdentity[]
+): Promise<[boolean, any[]]> {
+  if (presentation.type === MessageBodyType.SUBMIT_CLAIMS_FOR_CTYPES_PUBLIC) {
+    const attestedClaims = presentation.content.map(
+      AttestedClaim.fromAttestedClaim
+    )
+    const allVerified = await Promise.all(attestedClaims.map(ac => ac.verify()))
+    return [allVerified.every(b => b), attestedClaims]
```

Comment:
There might be a better way to check truthiness for all entries

Reply:
Very surprisingly, the fastest method to accomplish a result is actually `includes` here. I would propose to move this check to a separate function (like `noNulls` in _Claimer.ts_) and check with `includes`.

[Benchmark](https://jsben.ch/DSDaS)

Reply:
![Screenshot 2020-05-18 at 10 22 01](https://user-images.githubusercontent.com/14820950/82190654-a326f100-98f1-11ea-8ffd-7280b29e639d.png)

depends on the browser?

Reply:
Definitely, here is Safari:

<img width="594" alt="Screenshot 2020-05-18 at 10 38 15" src="https://user-images.githubusercontent.com/26296386/82192055-baff7480-98f3-11ea-84ef-64323d82e0f5.png">

Reply:
Obviously, we want to optimize for readability and not speed nuances. @weichweich and I decided to go with `includes`.  But definitely interesting to see the difference in browsers.

Reply:
Resolved in 6b71295